### DISCUSSION TITLE
[react-aria-components] - cannot style parent elements based on Input's data attributes

### DISCUSSION THREAD

**Discussion by binaryartifex** [adobe/react-spectrum#4770] [Q&A]
for the life of me i cannot figure out how to style a containing element based on the data attributes of the input component. use case -> icons or buttons that are meant to visually look like they are within the input field.

```
<TextField>
  <Label>First name</Label>
  <div className="wrapper">
  <Icon />
  <Input />
  </div>
  <Description />
</TextField>
```

so the use case here is that the input itself doesnt have a border or a background. its passed to the 'wrapper' class based on the data attribute or validity status of the input field, this is also so contextual styles can be passed to the icon as well. for the life of me i can't figure out how to do it.

**Comment by LFDanLu** [comment]
Is something like [this](https://codesandbox.io/s/boring-worker-smtrq9?file=/src/styles.css) what you are trying to accomplish? I've applied a border + background to the input directly for simplicity, styling them and the icon in accordance with the validity state will be much easier once https://github.com/adobe/react-spectrum/pull/4616 is released and the textfield wrapper itself has `data-validation-state` applied

**Reply by binaryartifex** [reply]
yeah 100% the only difference in implementations is lifting the styling responsibilities to a parent based specifically on validation state. although im a fan of using the hooks, it is kinda nice on the brain to see some lightweight ui components with all the 'hookery' abstracted away. Any idea as to when #4616 will be let loose on the world?

**Reply by LFDanLu** [reply]
It should go out with our next release which I'd say is tentatively in a couple weeks hopefully.
